src/Microsoft.OData.Core/ODataMediaTypeResolver.cs

```diff
@@ -41,41 +41,23 @@ public class ODataMediaTypeResolver
         {
             {
                 ODataPayloadKind.Batch,
-                new List<ODataMediaTypeFormat>
-                {
-                    new ODataMediaTypeFormat(new ODataMediaType(MimeConstants.MimeMultipartType, MimeConstants.MimeMixedSubType), ODataFormat.Batch)
-                }
+                new [] { new ODataMediaTypeFormat(new ODataMediaType(MimeConstants.MimeMultipartType, MimeConstants.MimeMixedSubType), ODataFormat.Batch) }
             },
             {
                 ODataPayloadKind.Value,
-                new List<ODataMediaTypeFormat>
-                {
-                    new ODataMediaTypeFormat(new ODataMediaType(MimeConstants.MimeTextType, MimeConstants.MimePlainSubType), ODataFormat.RawValue)
-                }
+                new [] { new ODataMediaTypeFormat(new ODataMediaType(MimeConstants.MimeTextType, MimeConstants.MimePlainSubType), ODataFormat.RawValue) }
             },
             {
                 ODataPayloadKind.BinaryValue,
-                new List<ODataMediaTypeFormat>
-                {
-                    new ODataMediaTypeFormat(new ODataMediaType(MimeConstants.MimeApplicationType, MimeConstants.MimeOctetStreamSubType), ODataFormat.RawValue)
-                }
+                new [] { new ODataMediaTypeFormat(new ODataMediaType(MimeConstants.MimeApplicationType, MimeConstants.MimeOctetStreamSubType), ODataFormat.RawValue) }
             },
             {
                 ODataPayloadKind.MetadataDocument,
-                new List<ODataMediaTypeFormat>
-                {
-                    new ODataMediaTypeFormat(new ODataMediaType(MimeConstants.MimeApplicationType, MimeConstants.MimeXmlSubType), ODataFormat.Metadata)
-                }
+                new [] { new ODataMediaTypeFormat(new ODataMediaType(MimeConstants.MimeApplicationType, MimeConstants.MimeXmlSubType), ODataFormat.Metadata) }
             },
             {
                 ODataPayloadKind.Asynchronous,
-                new List<ODataMediaTypeFormat>
-                {
-                    new ODataMediaTypeFormat(new ODataMediaType(MimeConstants.MimeApplicationType, MimeConstants.MimeHttpSubType), ODataFormat.RawValue)
-                }
-            },
-            {
-                ODataPayloadKind.Delta, new List<ODataMediaTypeFormat>()

```

Comment:
This does not look right. Maybe omitting this is causing the test case to be stuck forever.

Reply:
The original is wrong. ODataPayloadKind.Delta should same as ODataPayloadKind.Resource.  #Resolved.